pdm/openapi/action.yml

```
@@ -0,0 +1,115 @@
+name: Generate OpenAPI
+description: Generate and compare OpenAPI specifications
+
+inputs:
+  python-version:
+    description: Python version used to build
+    required: false
+    default: "3.11"
+  github-token:
+    description: A Github token with proper permissions
+    required: false
+    default: ${{ github.token }}
+  pypi-token:
+    description: Private PyPI token (JFrog read)
+    required: false
+  init:
+    description: Clone & sync
+    default: 'true'
+  group:
+    description: Dependency group(s) to install
+    default: docs
+  exclude-group:
+    description: Dependency group(s) to exclude from install
+    default: ''
+  spectral-file-glob:
+    description: Glob for Spectral to run on
+    default: 'docs/openapi.yaml'
+  spectral-ruleset:
+    description: Ruleset to enforce
+    default: '.github/openapi.spectral.yaml'
+
+runs:
+  using: composite
+  steps:
+  - name: Clone and install dependencies
+    uses: LedgerHQ/actions/pdm/init@main
+    id: init
+    if: inputs.init == 'true'
+    with:
+      group: ${{ inputs.group }}
+      exclude-group: ${{ inputs.exclude-group }}
+      pypi-token: ${{ inputs.pypi-token }}
+      python-version: ${{ inputs.python-version }}
+
+  - name: Generate the OpenAPI specifications
+    run: |
+      : Generate the OpenAPI specifications
+      pdm doc:openapi
+    env:
+      FORCE_COLOR: 'true'
+    shell: bash
+
+  - name: Run Spectral lint
+    uses: stoplightio/spectral-action@latest
+    with:
+      file_glob: ${{ inputs.spectral-file-glob }}
+      spectral_ruleset: ${{ inputs.spectral-ruleset }}
+      args: "--fail"
+
+  - name: Upload the OpenAPI specifications
+    uses: actions/upload-artifact@v4
+    with:
+      name: openapi
+      path: docs/openapi.yaml
+
+  - name: Move current OpenAPI specs to 'head'
+    if: steps.init.outputs.is_pr == 'true'
+    continue-on-error: true
+    run: |
+      mv docs/openapi.yaml openapi.head.yaml
+    shell: bash
+
+  - name: Checkout base ref
+    id: checkout-base
+    if: success() && steps.init.outputs.is_pr == 'true'
+    continue-on-error: true
+    run: |
+      CURRENT_REF=$(git rev-parse HEAD)
+      echo "backup-ref=${CURRENT_REF}" >> $GITHUB_OUTPUT
+
+      git restore .
+      git checkout ${{ github.base_ref }}
+    shell: bash
+
+  - name: Synchronize dependencies (base ref)
+    if: success() && steps.init.outputs.is_pr == 'true'
+    continue-on-error: true
+    run: |
+      pdm sync
+    shell: bash
```

```suggestion
  - name: Synchronize dependencies (base ref)
    if: success() && steps.init.outputs.is_pr == 'true'
    continue-on-error: true
    run: |
      pdm syncenv:
    env:
        PYPI_DEPLOY_TOKEN: ${{ inputs.pypi-token || ''  }}
    shell: bash
```
I think here you need something like this again?

---

I'm checking again but I don't think we need to specify the token again as the HEAD is merged into the base ref so the token stay the same